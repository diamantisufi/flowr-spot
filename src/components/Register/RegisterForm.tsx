import React from 'react'
import { TextInput } from 'components/TextInput'
import { ErrorComponent } from 'components/Error'

type RegisterFormProps = {
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void
  error: string
}

export const RegisterForm: React.FC<RegisterFormProps> = ({
  error,
  onSubmit,
}) => {
  return (
    <div className="flex flex-col">
      <form onSubmit={onSubmit}>
        <div className="mx-auto flex flex-col items-center justify-center gap-3">
          <div className="flex justify-between gap-[10px] w-full">
            <TextInput
              label="First Name"
              type="text"
              name="first_name"
              id="first_name"
              required
              className="w-full"
            />
            <TextInput
              label="Last Name"
              type="text"
              name="last_name"
              id="last_name"
              required
            />
          </div>
          <TextInput
            label="Date of Birth"
            type="date"
            name="date_of_birth"
            id="date_of_birth"
            required
          />
          <TextInput
            label="Email Address"
            type="email"
            name="email"
            id="email"
            required
          />
          <TextInput
            label="Password"
            type="password"
            name="password"
            id="password"
            minLength={6}
            required
          />
          {error && <ErrorComponent error={error} />}
          <button
            type="submit"
            className="button-primary w-full p-5 mt-3 text-white font-ubuntu font-medium"
          >
            Create Account
          </button>
        </div>
      </form>
    </div>
  )
}
