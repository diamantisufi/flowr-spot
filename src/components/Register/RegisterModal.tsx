import React, { useState } from 'react'
import Modal from 'react-modal'
import { modalStyles } from 'common/modalStyles'
import { RegisterForm } from './RegisterForm'
import { registerUser } from 'services/userService'
import { RegisterResponseData } from 'helpers/types'
import { ModalCloseIcon } from 'components/ModalClose'

type LoginModalProps = {
  isOpen: boolean
  onClose: () => void
  handleRegisterSuccess: () => void
}

export const RegisterModal: React.FC<LoginModalProps> = ({
  isOpen,
  onClose,
  handleRegisterSuccess,
}) => {
  const [error, setError] = useState<string>('')
  const [loading, setLoading] = useState<boolean>(false)

  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    const formData = new FormData(event.currentTarget)

    const formDataString = JSON.stringify(Object.fromEntries(formData))
    const payload = JSON.parse(formDataString)

    setLoading(true)
    await registerUser(payload)
      .then((response: RegisterResponseData | undefined) => {
        if (response?.auth_token) {
          handleRegisterSuccess()
        }
      })
      .catch((error) => {
        setError(error?.response?.data?.error)
      })
      .finally(() => {
        setLoading(false)
      })
  }

  return (
    <Modal
      isOpen={isOpen}
      contentLabel='Login Modal'
      style={modalStyles}
      shouldCloseOnOverlayClick
      shouldCloseOnEsc
      onRequestClose={onClose}
      ariaHideApp={false}
    >
      <div className='modal-container'>
        <ModalCloseIcon onClose={onClose} />
        <h2 className='text-gray font-medium font-ubuntu mb-7'>
          {error ? 'Error' : 'Create an Account'}
        </h2>
      </div>
      <RegisterForm onSubmit={handleSubmit} error={error} loading={loading} />
    </Modal>
  )
}
