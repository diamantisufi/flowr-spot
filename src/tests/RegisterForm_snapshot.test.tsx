import { RegisterForm } from 'components/Register/RegisterForm'
import React from 'react'
import renderer from 'react-test-renderer'

describe('RegisterForm Snapshot', () => {
  it('matches snapshot', () => {
    const mockOnSubmit = jest.fn()
    const mockError = ''

    const tree = renderer
      .create(
        <RegisterForm
          onSubmit={mockOnSubmit}
          error={mockError}
          loading={false}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
