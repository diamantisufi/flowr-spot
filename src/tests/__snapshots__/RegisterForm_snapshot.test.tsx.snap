// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Snapshot matches snapshot 1`] = `
<div
  className="flex flex-col"
>
  <form
    data-testid="register-form"
    onSubmit={[MockFunction]}
  >
    <div
      className="mx-auto flex flex-col items-center justify-center gap-3"
    >
      <div
        className="flex justify-between gap-[10px] w-full"
      >
        <div
          className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
        >
          <label
            className="absolute top-2 text-[10px] text-secondary ml-4"
            htmlFor="first_name"
          >
            First Name
          </label>
          <input
            className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 w-full"
            id="first_name"
            name="first_name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
        >
          <label
            className="absolute top-2 text-[10px] text-secondary ml-4"
            htmlFor="last_name"
          >
            Last Name
          </label>
          <input
            className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 undefined"
            id="last_name"
            name="last_name"
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
      >
        <label
          className="absolute top-2 text-[10px] text-secondary ml-4"
          htmlFor="date_of_birth"
        >
          Date of Birth
        </label>
        <input
          className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 undefined"
          id="date_of_birth"
          name="date_of_birth"
          required={true}
          type="date"
        />
      </div>
      <div
        className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
      >
        <label
          className="absolute top-2 text-[10px] text-secondary ml-4"
          htmlFor="email"
        >
          Email Address
        </label>
        <input
          className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 undefined"
          id="email"
          name="email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="relative rounded flex flex-col w-full h-full border border-inputBorder bg-inputBackground hover:border-slate-300 transition-colors text-[13px]"
      >
        <label
          className="absolute top-2 text-[10px] text-secondary ml-4"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="outline-none h-full w-full bg-transparent text-gray p-4 pt-7 undefined"
          id="password"
          minLength={6}
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="button-primary w-full p-5 mt-3 text-white font-ubuntu font-medium"
        type="submit"
      >
        Create Account
      </button>
    </div>
  </form>
</div>
`;
